.github/actions/setup-java-build/action.yml

```
@@ -0,0 +1,26 @@
+name: "Setup Java build"
+description: "Performs the setup of required build tools (eg.: Maven, Java)"
+inputs:
+  java-version:
+    description: the desired Java version
+    default: "17"
+    required: false
+  java-distribution:
+    description: the desired Java distribution
+    default: "temurin"
+    required: false
+runs:
+  using: composite
+  steps:
+    - name: "Cache local Maven repository"
+      uses: actions/cache@v2
```

```suggestion
      uses: actions/cache@v3
```

will also fix the deprecation notices you should have seen